packages/sync-repo-settings/src/sync-repo-settings.ts

```diff
@@ -64,6 +72,76 @@ const branchProtectionDefaults = deepFreeze({
  * Main.  On a nightly cron, update the settings for a given repository.
  */
 export function handler(app: Application) {
+
+  // Lint any pull requests that touch configuration
+  app.on(
+    [
+      'pull_request.opened',
+      'pull_request.reopened',
+      'pull_request.synchronize',
+    ],
+    async (context: Context) => {
+      const owner = context.payload.repository.owner.login;
+      const repo = context.payload.repository.name;
+      const number = context.payload.number;
+      const files = await context.github.paginate(
+        context.github.pulls.listFiles.endpoint.merge({
+          owner,
+          repo,
+          pull_number: number,
+          per_page: 100,
+        })
+      );
+      for (const file of files) {
+        if (
+          file.status === 'deleted' ||
+          (file.filename !== `.github/${configFileName}` &&
+          (repo !== '.github' || file.filename !== configFileName))
+        ) {
+          continue;
+        }
+        const blob = await context.github.git.getBlob({
+          owner,
+          repo,
+          file_sha: file.sha,
+        });
+        const configYaml = Buffer.from(blob.data.content, 'base64').toString('utf8');
+        const config = yaml.safeLoad(configYaml, {
+
+        });
+        let isValid = false;
+        let errorText = '';
+        if (typeof config === 'object') {
+          const validateSchema = ajv.compile(schema);
+          isValid = await validateSchema(config);
+          errorText = JSON.stringify(validateSchema.errors, null, 4);
+        } else {
+          errorText = 'sync-repo-settings.yaml is not valid YAML 😱'
+        }
+
+        const checkParams: Octokit.ChecksCreateParams = context.repo({
+          name: 'sync-repo-settings-check',
+          head_sha: context.payload.pull_request.head.sha,
+          conclusion: 'success',
+        });
+        if (!isValid) {
+          checkParams.conclusion = 'failure',
+          checkParams.output = {
+            title: 'Invalid sync-repo-settings.yaml schema 😱',
```

Comment:
Can we get some sort of nice error message from the validation? Looking at https://ajv.js.org/#validation-errors, perhaps the `message` field?

Reply:
Yes!  So the issue is there can be many errors.  For now I created a detail view with the JSON.stringified text of all errors, and use it as the full error text for folks that dig in.